### PR TITLE
Refactored richestusers, atm and resetsymbol

### PR DESCRIPTION
**Pull request by SilverTactic**

### CODE REVIEW

**Comment by panpawn** [comment]
I don't like the changes to the ATM command at all...

I'm going to review the rest closer later, but they seem more complicated than they have to be.

File: chat-plugins/shop.js [hunk 1]
```diff
@@ -348,73 +348,31 @@ exports.commands = {
...
-			}
-			return output;
-		}
-		return this.sendReplyBox(atm(target));
+		target = (Users.getExact(target) ? Users.getExact(target).name : target);
+		var bucks = economy.readMoney(toId(target)) || 'does not have any';
+		if (parseInt(bucks)) bucks = 'has ' + bucks;
+		var output = '<u>Gold Wallet:</u><br />' + 
+			'<b><span style = "color:' + Gold.hashColor(toId(target)) + '">' + Tools.escapeHTML(target) + '</span></b> ' + bucks + ' Gold ' +
+			(bucks === 1 ? 'buck' : 'bucks');
```
Comment (panpawn) [inline chat-plugins/shop.js]: Can we put this on the line above pls

File: chat-plugins/shop.js [hunk 2]
```diff
@@ -348,73 +348,31 @@ exports.commands = {
...
-		for (var i = row.length; i > -1; i--) {
-			if (!row[i]) continue;
-			var parts = row[i].split(",");
-			userids.id[i] = parts[0];
-			userids.money[i] = Number(parts[1]);
-			size++;
-			if (isNaN(parts[1]) || parts[1] === 'Infinity') userids.money[i] = 0;
-		}
+		var buckList = fs.readFileSync('config/money.csv', 'utf8')
+			.split('\n')
```
Comment (panpawn) [inline chat-plugins/shop.js]: Put this on line above

File: chat-plugins/shop.js [hunk 3]
```diff
@@ -348,73 +348,31 @@ exports.commands = {
...
+			.map(function (line) {
```
Comment (panpawn) [inline chat-plugins/shop.js]: And this

File: chat-plugins/shop.js [hunk 4]
```diff
@@ -348,73 +348,31 @@ exports.commands = {
...
+				return line.split(',');
+			})
+			.sort(function (a, b) {
```
Comment (panpawn) [inline chat-plugins/shop.js]: And this

File: chat-plugins/shop.js [hunk 5]
```diff
@@ -486,14 +444,9 @@ exports.commands = {
 	rs: 'resetsymbol',
 	resetsymbol: function(target, room, user) {
 		if (!user.hasCustomSymbol) return this.sendReply('You don\'t have a custom symbol!');
-		user.getIdentity = function() {
-			if (this.muted) return '!' + this.name;
-			if (this.locked) return '‽' + this.name;
-			return this.group + this.name;
-		};
-		user.hasCustomSymbol = false;
-		delete user.getIdentity;
```
Comment (panpawn) [inline chat-plugins/shop.js]: Why are we removing this line?

**Comment by panpawn** [comment]
Gave more input @SilverTactic

File: chat-plugins/shop.js [hunk 6]
```diff
@@ -348,73 +348,31 @@ exports.commands = {
...
-					output += "<b><font color=\"" + Gold.hashColor(target) + "\">" + originalName + "</font></b> has " + bucks + label + ".";
-			}
-			return output;
-		}
-		return this.sendReplyBox(atm(target));
+		target = (Users.getExact(target) ? Users.getExact(target).name : target);
+		var bucks = economy.readMoney(toId(target)) || 'does not have any';
+		if (parseInt(bucks)) bucks = 'has ' + bucks;
+		var output = '<u>Gold Wallet:</u><br />' + 
+			'<b><span style = "color:' + Gold.hashColor(toId(target)) + '">' + Tools.escapeHTML(target) + '</span></b> ' + bucks + ' Gold ' +
```
Comment (panpawn) [inline chat-plugins/shop.js]: ``Gold.hashColor();`` automatically takes the id, this ``toId();`` is not needed - please remove it

**Comment by panpawn** [comment]
Also, it would be cool for richest user if it showed the colors with their username color

File: chat-plugins/shop.js (hunk 5 above)
Comment (SilverTactic) [inline chat-plugins/shop.js]: Check the lines I added after this. <code>user.getIdentity = Users.User.prototype.getIdentity;</code> resets the definition of <code>user.getIdentity()</code> to what it's originally supposed to be (we change the definition of this in the customsymbol command).

**Comment by SilverTactic** [comment]
Finished~

**Comment by panpawn** [comment]
Said this on the server, but I thought I might as well remind you here, too...

For the richestuser command, let's make it so that it:
- Is in the form of a table with columns of ``Position:``, ``User:``, ``Bucks:``
- Displays the top 10 by default (only broadcastable for the top 10, will default to the top 10 if broadcasted  otherwise)
- Can do things like ``/richestuser 15`` to shop the top 15 richest users
- Can do things like ``/richestuser panpawn`` to see what position someone is (is broadcastable)

**Comment by panpawn** [comment]
Any updates on the progress on richestuser's command functionality?

**Comment by SilverTactic** [comment]
Nope, not yet. Still need to work on it

**Comment by panpawn** [comment]
These have been changed since and there's merge conflicts, so closing this for now.

**Pull request closed without merging**